logic/transactions/pool.js

```diff
@@ -0,0 +1,922 @@
+ * @private
+ * @param {transaction} transaction
+ * @param {Object} poolList
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+__private.addToPoolList = function (transaction, poolList, cb) {
+	if (poolList.transactions[transaction.id] === undefined) {
+		poolList.count++;
+	}
+	poolList.transactions[transaction.id] = transaction;
+	return setImmediate(cb);
+};
+
+/**
+ * Adds transactions to pool list without checks.
+ * @private
+ * @param {transaction} transaction
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+__private.addReadyCheckBroadcast = function (transaction, cb) {
+	if (transaction.broadcast) {
+		delete transaction.broadcast;
+		pool.broadcast.push(transaction);
+	}
+	__private.addToPoolList(transaction, pool.verified.ready, cb);
+};
+
+/**
+ * Creates signature based on multisignature transaction and secret.
+ * @private
+ * @param {transaction} transaction
+ * @param {String} secret
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+__private.createSignature = function (transaction, secret, cb) {
+	var hash = crypto.createHash('sha256').update(secret, 'utf8').digest();
+	var keypair = library.ed.makeKeypair(hash);
+	var publicKey = keypair.publicKey.toString('hex');
+	
+	if (transaction.asset.multisignature.keysgroup.indexOf('+' + publicKey) === -1) {
+		return setImmediate(cb, 'Permission to sign transaction denied');
+	}
+	var signature = library.logic.transaction.multisign(keypair, transaction);
+	return setImmediate(cb, null, signature);
+};
+
+/**
+ * Adds transaction id to invalid pool list.
+ * @private
+ * @param {string} id
+ */
+__private.addInvalid = function (id) {
+	pool.invalid.transactions[id] = true;
+	pool.invalid.count++;
+};
+
+/**
+ * Deletes id from pool list.
+ * @private
+ * @param {string} id
+ * @param {Object} poolList
+ * @return {boolean} true if transaction id is on the list and was deleted
+ */
+__private.delete = function (id, poolList) {
+	var transaction = poolList.transactions[id];
+
+	if (transaction !== undefined) {
+		delete poolList.transactions[id];
+		poolList.count--;
+		return true;
+	}
+	return false;
+};
+
+/**
+ * Sums unverified, verified.pending and verified.ready counters.
+ * @private
+ * @return {Number} Total = unverified + pending + ready
+ */
+__private.countTransactionsPool = function () {
+	return pool.unverified.count + pool.verified.pending.count + pool.verified.ready.count;
+};
+
+/**
+ * Removes transactions if expired from pool list.
+ * @private
+ * @implements {__private.transactionTimeOut}
+ * @implements {__private.delete}
+ * @param {Object[]} poolList
+ * @param {function} cb - Callback function
+ * @return {setImmediateCallback} cb|error
+ */
+__private.expireTransactionsFromList = function (poolList, cb) {
+	async.eachSeries(poolList.transactions, function (transaction, eachSeriesCb) {
```

Comment:
why eachSeries?

Reply:
You are right, expiration can be done it in parallel with `async.each`